### PR TITLE
Ai fix2

### PR DESCRIPTION
**Pull request by gugus**
AiFollow and AiFollowCell are implemented

### CODE REVIEW

**Comment by zinnschlag** [comment]
Something about the code for AiFollow, AiEscort and AiCombat is really weird. You are setting local xCell/yCell variables to a certain value but never make use of these variables. Also, you have member variables cellX and cellY in these classes, that violate our naming guidelines.

**Comment by gugus** [comment]
yeah, the implementation of pathfinding was changed at some point. Need a (little?) clean up

File: apps/openmw/mwmechanics/aifollow.cpp [hunk 1]
```diff
@@ -15,13 +19,100 @@
...
+        mStuckPos = pos;
+    }
+
+    if(mPathFinder.getPath().empty())
+        mPathFinder.addPointToPath(dest);
+
+    if(!mPathFinder.checkPathCompleted(pos.pos[0],pos.pos[1],pos.pos[2]))
+    {
+        float zAngle = mPathFinder.getZAngleToNext(pos.pos[0], pos.pos[1]);
+        MWBase::Environment::get().getWorld()->rotateObject(actor, 0, 0, zAngle, false);
```
Comment (ghost) [inline apps/openmw/mwmechanics/aifollow.cpp]: Please do not use rotateObject directly. That will cause funny effects when being in knockdown or paralyzed or other states where the actor is not supposed to move. Instead, use the movement settings. 
(Other AI packages currently suffer from this problem too, which is why I added todo comments there)

Comment (gugus) [inline apps/openmw/mwmechanics/aifollow.cpp]: ok, I'll change it.

**Comment by zinnschlag** [comment]
I am not entirely happy with the casting. That could have easily been done with a polymorphic approach. But considering were we standing on our way to 1.0, we can as well go along with it for now and polish it up after 1.0.
I will do a bit of cleanup and then merge.

**Pull request merged**